### ISSUE TITLE
1.3-beta1 BUG - Pictures stored on an SD card cannot be deleted

### ISSUE THREAD

**Issue by haladam** [franciscofranco/Focus-bugtracker#13]
Attempting to delete pictures stored on an sd card only "hides" them both from Focus and the stock gallery. After a restart of the device, they can be found untouched in both the stock gallery and focus.

Samsung Galaxy S7, latest marshmallow build, completely stock firmware. SD card is encrypted.

**Comment by franciscofranco** [comment]
Are you using Knox?

**Comment by haladam** [comment]
No, I've never used it - no need for it.

**Comment by franciscofranco** [comment]
Interesting, I managed to reproduce on my One Plus X. I'll see what I can do.

**Comment by haladam** [comment]
This is a screen recording of what I meant with asking for the specific SD card permission, the way Solid explorer does it:
https://drive.google.com/file/d/0BzXAtOEzQdO5clJmaU16cXlBanc/view?usp=drivesdk

Perhaps it's something that's needed? I'm an amateur user but I've never had problems with deleting anything on an SD card with solid explorer, so maybe it helps here as well

**Comment by vseehua** [comment]
I cannot put photos on sd card into a vault as well. Maybe focus cannot write properly to the sd card?

**Comment by franciscofranco** [comment]
Yeah I know what it is. I truthfully didn't know I needed to use Android's DocumentFile provider to delete/move files from the external sdcard. I'll work on this.

Focus can work just fine with the internal sdcard, but external sdcard support is ridiculously retarded on Android.

**Comment by haladam** [comment]
Well, maybe not on Android in general (when the SD card is formatted as internal storage perhaps the problem is not there?) but since Samsung decided for its own implementation (skipping formatting it as internal storage completely), it causes problems for apps now. However, a simple user just as myself will blame the problem on the app :-)

**Comment by OmerFarah** [comment]
I have the same issue here. pics taken with camera which are been saved to SD card. Im using note 3 neo running on aicp ROM.

**Comment by franciscofranco** [comment]
@haladam nothing to do with that, it's just that since Lollipop we need to use a special thing called Document provider to edit external sdcard content. I didn't know about it so I just assumed the same mechanism worked for both normal storage & external.

**Comment by haladam** [comment]
Oh, well. At least the problem and the solution (workaround) are known.
